sources/api/pluto/src/api.rs

```diff
@@ -0,0 +1,101 @@
+pub(super) use inner::{get_aws_k8s_info, Error};
+
+/// The result type for the [`api`] module.
+pub(super) type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Default)]
+pub(crate) struct AwsK8sInfo {
+    pub(crate) region: String,
+    pub(crate) cluster_name: String,
+}
+
+/// This code is the 'actual' implementation compiled when the `sources` workspace is being compiled
+/// for `aws-k8s-*` variants.
+// TODO - find a better way https://github.com/bottlerocket-os/bottlerocket/issues/1260
+#[cfg(aws_k8s_variant)]
+mod inner {
+    use super::*;
+    use snafu::{OptionExt, ResultExt, Snafu};
+
+    // FIXME Get these from configuration in the future
+    const DEFAULT_API_SOCKET: &str = "/run/api.sock";
+    const SETTINGS_URI: &str = "/settings";
+
+    #[derive(Debug, Snafu)]
+    pub(crate) enum Error {
+        #[snafu(display("Error calling Bottlerocket API: {}", source))]
+        ApiClient {
+            source: apiclient::Error,
+            uri: String,
+        },
+
+        #[snafu(display("AWS settings are missing"))]
+        AwsSettings {},
+
+        #[snafu(display("The 'aws.region' setting is missing"))]
+        AwsRegionMissing,
```

Comment:
nit: a similar `Missing { missing: String }` could simplify this; the conditional modules are unfortunate and minimizing them seems helpful.  Could also pull the non-`fn` bits out to share..?